src/geometry/algorithms/ArborX_Centroid.hpp

```diff
@@ -40,7 +40,7 @@ using namespace GeometryTraits;
 template <typename Point>
 struct centroid<PointTag, Point>
 {
-  KOKKOS_FUNCTION static constexpr auto apply(Point const &point)
+  KOKKOS_FUNCTION static constexpr auto &apply(Point const &point)
```

Comment:
```suggestion
  KOKKOS_FUNCTION static constexpr Point &apply(Point const &point)
```

Reply:
Ping

Reply:
I did not see how it made any difference.